src/lib.rs

```diff
@@ -1283,52 +1283,46 @@ impl CecConnection {
         }
     }
 
-    pub fn volume_up(&self, send_release: bool) -> CecConnectionResult<()> {
-        if unsafe { libcec_volume_up(self.1, send_release.into()) } == 0 {
-            Err(CecConnectionResultError::TransmitFailed)
-        } else {
-            Ok(())
+    pub fn volume_up(
+        &self,
+        send_release: bool,
+    ) -> Result<KnownCecAudioStatus, TryFromCecAudioStatusError> {
+        match u8::try_from(unsafe { libcec_volume_up(self.1, send_release.into()) }) {
+            Ok(status) => KnownCecAudioStatus::try_from(status),
+            Err(_) => Err(TryFromCecAudioStatusError::Unknown),
         }
     }
 
-    pub fn volume_down(&self, send_release: bool) -> CecConnectionResult<()> {
-        if unsafe { libcec_volume_down(self.1, send_release.into()) } == 0 {
-            Err(CecConnectionResultError::TransmitFailed)
-        } else {
-            Ok(())
+    pub fn volume_down(
+        &self,
+        send_release: bool,
+    ) -> Result<KnownCecAudioStatus, TryFromCecAudioStatusError> {
+        match u8::try_from(unsafe { libcec_volume_down(self.1, send_release.into()) }) {
+            Ok(status) => KnownCecAudioStatus::try_from(status),
+            Err(_) => Err(TryFromCecAudioStatusError::Unknown),
         }
     }
 
-    pub fn mute_audio(&self, send_release: bool) -> CecConnectionResult<()> {
-        if unsafe { libcec_mute_audio(self.1, send_release.into()) } == 0 {
-            Err(CecConnectionResultError::TransmitFailed)
-        } else {
-            Ok(())
+    pub fn mute_audio(
+        &self,
+        send_release: bool,
+    ) -> Result<KnownCecAudioStatus, TryFromCecAudioStatusError> {
+        match u8::try_from(unsafe { libcec_mute_audio(self.1, send_release.into()) }) {
+            Ok(status) => KnownCecAudioStatus::try_from(status),
+            Err(_) => Err(TryFromCecAudioStatusError::Unknown),
```

Comment:
`volume_up`, `volume_down`, & `mute_audio` are pretty weird. There's some ambiguity that makes it impossible to distinguish errors, and a successful fallback that changes the TV's volume when you don't have an AVR connected.
  
Basically, it's not possible to query the audio status of a TV, so the library will return "unknown" in that case... but "unknown" is used as a catch-all for errors.

The way it's implemented right now though, these functions will always return an error. Unless you have an AVR that ends up with an unmuted volume of 0 after the function call.